diff --git a/nan.h b/nan.h
index 9e7c59e488497c4d6b93af69176b08385565745b..0326246aa9a308b57f719f7334c24170b9dd930c 100644
--- a/nan.h
+++ b/nan.h
@@ -2516,8 +2516,7 @@ inline void SetAccessor(
   , SetterCallback setter = 0
   , v8::Local<v8::Value> data = v8::Local<v8::Value>()
   , v8::AccessControl settings = v8::DEFAULT
-  , v8::PropertyAttribute attribute = v8::None
-  , imp::Sig signature = imp::Sig()) {
+  , v8::PropertyAttribute attribute = v8::None) {
   HandleScope scope;
 
   imp::NativeGetter getter_ =
@@ -2550,9 +2549,6 @@ inline void SetAccessor(
     , obj
     , settings
     , attribute
-#if (NODE_MODULE_VERSION < NODE_18_0_MODULE_VERSION)
-    , signature
-#endif
   );
 }
 
diff --git a/nan_callbacks.h b/nan_callbacks.h
index 53ede846ac9a865a737218dabbbd48305d3d6b63..85b2ca8c7da2fa332d35b90b775fbc14010601a1 100644
--- a/nan_callbacks.h
+++ b/nan_callbacks.h
@@ -52,7 +52,6 @@ typedef void(*IndexQueryCallback)(
     const PropertyCallbackInfo<v8::Integer>&);
 
 namespace imp {
-typedef v8::Local<v8::AccessorSignature> Sig;
 
 static const int kDataIndex =                    0;
 